### PR TITLE
improved debugging techniques

### PR DESCRIPTION
**Pull request by goldhoorn**

### CODE REVIEW

File: src/documentation/debugging/index.page [hunk 1]
```diff
@@ -7,7 +7,10 @@ Introduction
 ----------------
 
 Rock provides a logger that facilitates and unifies debugging of system libraries. 
-The following section describes the functionality of the base logger and how to use it. 
+The following section describes the functionality of the base logger and how to use it.
+
+For debugging of running/existing components please refer [Debugging Components](../runtime/debugging.html)
```
Comment (doudou) [inline src/documentation/debugging/index.page]: please refer to

File: src/documentation/runtime/debugging.page [hunk 2]
```diff
@@ -72,7 +72,31 @@ Orocos.run 'my_project::Task' => 'task', 'my_deployment',
...
 in which case the valgrind output is redirected to a separate process logfile,
 which is the process' log file name with .valgrind appended (in the example
 above, task-PID.log.valgrind and my_deployment-PID.log.valgrind, where PID is
 the process ID of the started deployment)
+
+It is also possible to pass options to valgrind:
+
+~~~ ruby
+Orocos.run 'my_project::Task' => 'task', 'my_deployment',
+    :valgrind => true, :valgrind_options => ['--undef-value-errors=no'] do
```
Comment (doudou) [inline src/documentation/runtime/debugging.page]: Use the option: value syntax

File: src/documentation/runtime/debugging.page [hunk 3]
```diff
@@ -72,7 +72,31 @@ Orocos.run 'my_project::Task' => 'task', 'my_deployment',
...
+   ...
+end
+~~~
+
+Ignore RTT Exception Handling
+--------
+RTT catches by default all exeption from the tasks. 
+This is because RTT tries to be as safe as possible.
+A somehow crashed task should never influence the execution of
+other components within the same compositon. However for debugging
```
Comment (doudou) [inline src/documentation/runtime/debugging.page]: composition => deployment

File: src/documentation/runtime/debugging.page [hunk 4]
```diff
@@ -72,7 +72,31 @@ Orocos.run 'my_project::Task' => 'task', 'my_deployment',
...
+it is usefull to let a component raise the exception to cause a crash.
```
Comment (doudou) [inline src/documentation/runtime/debugging.page]: However, for debugging purposes, it is convenient to let the exceptions be unhandled and cause a process abort.

File: src/documentation/runtime/debugging.page [hunk 5]
```diff
@@ -72,7 +72,31 @@ Orocos.run 'my_project::Task' => 'task', 'my_deployment',
...
+
+Ignore RTT Exception Handling
+--------
+RTT catches by default all exeption from the tasks. 
+This is because RTT tries to be as safe as possible.
+A somehow crashed task should never influence the execution of
+other components within the same compositon. However for debugging
+it is usefull to let a component raise the exception to cause a crash. 
+This makes the debugging with gdb easier. To activate this **debugging** behaviour
+set the env RTT_IGNORE_EXCEPTION=1
```
Comment (doudou) [inline src/documentation/runtime/debugging.page]: set the environment variable RTT_IGNORE_EXCEPTION to 1

File: src/documentation/runtime/debugging.page (hunk 2 above)
Comment (goldhoorn) [inline src/documentation/runtime/debugging.page]: What did you mean by that?

**Comment by goldhoorn** [comment]
expect the one remainig question: updated

**Comment by goldhoorn** [comment]
pong

**Pull request merged**